packages/vow/test/unhandled-rejections.test.js

```diff
@@ -0,0 +1,274 @@
+import {
+  annihilate,
+  startLife,
+  test,
+} from '@agoric/swingset-vat/tools/prepare-strict-test-env-ava.js';
+
+import { Far } from '@endo/far';
+import { makeDurableZone } from '@agoric/zone/durable.js';
+
+import { makeExpectUnhandledRejectionMacro } from '@agoric/internal/src/lib-nodejs/ava-unhandled-rejection.js';
+import { makeGcAndFinalize } from '@agoric/internal/src/lib-nodejs/gc-and-finalize.js';
+import engineGC from '@agoric/internal/src/lib-nodejs/engine-gc.js';
+import { prepareVowTools } from '../vat.js';
+
+const expectUnhandled = test.macro(
+  makeExpectUnhandledRejectionMacro(import.meta.url),
+);
+
+const gcAndFinalize = makeGcAndFinalize(engineGC);
+test.afterEach.always(gcAndFinalize);
+
+const shorten = vow => vow.payload.vowV0.shorten();
+
+const DEFAULT_REASON = Error('some kind of rejection');
+
+const makeRejectedVow = ({ makeVowKit }, reason = DEFAULT_REASON) => {
+  const { resolver, vow } = makeVowKit();
+  resolver.reject(reason);
+  return vow;
+};
+
+/**
+ * A vow on which `shorten` is called, and is then rejected in the same
+ * incarnation (baseline, should not trigger the unhandled rejection handler)
+ */
+test.serial('vow shortened, then rejected', async t => {
+  annihilate();
+
+  await startLife(async baggage => {
+    const zone = makeDurableZone(baggage, 'durableRoot');
+    const vowTools = prepareVowTools(zone);
+
+    const testVowKit = vowTools.makeVowKit();
+
+    shorten(testVowKit.vow).then(
+      value => t.fail(value),
+      reason =>
+        t.throws(
+          () => {
+            throw reason;
+          },
+          { message: DEFAULT_REASON.message },
+        ),
+    );
```

Comment:
Why not the following?

At the very least I believe the test needs to await something, likely the promise returned by `.then`.

Applies to all test cases here

```suggestion
    await t.throwsAsync(
      () => shorten(testVowKit.vow),
      { message: DEFAULT_REASON.message },
    );
```

Reply:
Terrific!  Thanks for pointing out my misuse of AVA. :)